### PR TITLE
Convert ProductLink tab into web component

### PR DESCRIPTION
**Pull request by tvdeyen**
Alchemy 7.2 [converted the link dialog into a view component](https://github.com/AlchemyCMS/alchemy_cms/pull/2796).
The current view partials and javascript does not work anymore.

Adding a `alchemy-product-select` web component.

### CODE REVIEW

**Comment by mamhoff** [comment]
This looks all good to me, but the tests are very broken...

**Comment by tvdeyen** [comment]
Yes, but totally unrelated.

File: lib/generators/alchemy/solidus/install/install_generator.rb [hunk 1]
```diff
@@ -123,25 +123,25 @@ def inject_routes
...
-             yes?("\nDo you want Alchemy to handle the root route '/'? (y/n)")
+            yes?("\nDo you want Alchemy to handle the root route '/'? (y/n)")
           sentinel = "Rails.application.routes.draw do\n"
-          inject_into_file routes_file_path, { after: sentinel } do
+          inject_into_file routes_file_path, {after: sentinel} do
             <<~ROOT_ROUTE
-              \  # Let AlchemyCMS handle the root route
-              \  root to: 'alchemy/pages#index'
+              # Let AlchemyCMS handle the root route
+              root to: 'alchemy/pages#index'
```
Comment (mamhoff) [inline lib/generators/alchemy/solidus/install/install_generator.rb:144]: This is the change that causes the specs to fail. Without the leading backslashes, the two lines have the wrong indentation, and the install generator does not realize the string is already there, leading to `root` being called twice.

Comment (tvdeyen) [reply lib/generators/alchemy/solidus/install/install_generator.rb:144]: WOW! Good eyes

Comment (tvdeyen) [reply lib/generators/alchemy/solidus/install/install_generator.rb]: Extracted the Rubocop changes into dedicated PR

Comment (mamhoff) [reply lib/generators/alchemy/solidus/install/install_generator.rb]: > WOW! Good eyes

Nah I didn't *see* that. I debugged it :)

File: Gemfile [hunk 2]
```diff
@@ -11,7 +11,7 @@ end
 
 alchemy_version = ENV.fetch("ALCHEMY_VERSION", "main")
 if alchemy_version == "main"
-  gem "alchemy_cms", github: "AlchemyCMS/alchemy_cms", branch: "main"
+  gem "alchemy_cms", github: "AlchemyCMS/alchemy_cms", branch: "admin_importmaps"
```
Comment (mamhoff) [inline Gemfile:14]: We can now do `main`! (For when you rebase the conflict in the `ci.yml`).

Comment (tvdeyen) [reply Gemfile]: done

**Pull request merged**